File: ruby/simplify_path.rb
# @param {String} path
# @return {String}
def simplify_path(path)
  stack = []
  for item in path.split("/")
    if ["", "."].include? item
      next
    elsif item == ".."
      stack.pop if stack
    else
      stack << item
    end
  end
  return "/" + stack.join("/")
end

File: python/simplifyPath.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        stack = []
        for item in path.split("/"):
            if item in ["", "."]:
                next
            elif item == "..":
                if stack:
                    stack.pop() 
            else:
                stack.append(item)
        return "/" + "/".join(stack)
